react/ProductSummaryList.js

```diff
@@ -0,0 +1,193 @@
+import { mapCatalogProductToProductSummary } from './utils/normalize'
+
+import productsQuery from './graphql/products.graphql'
+
+const ProductSummaryListStateContext = createContext(undefined)
+
+const ProductSummaryList = ({ children, data, intl, buyButtonText }) => {
+  const formattedBuyButtonText = formatIOMessage({ id: buyButtonText, intl })
+  const componentList = data.products.map(product => {
+    const normalizedProduct = mapCatalogProductToProductSummary(product)
+
+    return (
+      <ProductSummary key={product.productId} product={normalizedProduct}>
+        <ProductSummaryImage />
+        <ProductSummaryName />
+        <Spacer />
+        <ProductSummaryPrice />
+        <ProductSummaryBuyButton buyButtonText={formattedBuyButtonText} />
+      </ProductSummary>
+    )
+  })
+
+  return (
+    <ProductSummaryListStateContext.Provider value={componentList}>
+      {children}
+    </ProductSummaryListStateContext.Provider>
+  )
+}
+
+function useProductSummaryListState() {
+  const context = useContext(ProductSummaryListStateContext)
+  return context
+}
+
+const parseFilters = ({ id, value }) => `specificationFilter_${id}:${value}`
+
+const ORDER_BY_OPTIONS = {
+  ORDER_BY_RELEVANCE: {
+    name: 'admin/editor.productSummaryList.orderType.relevance',
+    value: '',
+  },
+  ORDER_BY_TOP_SALE_DESC: {
+    name: 'admin/editor.productSummaryList.orderType.sales',
+    value: 'OrderByTopSaleDESC',
+  },
+  ORDER_BY_PRICE_DESC: {
+    name: 'admin/editor.productSummaryList.orderType.priceDesc',
+    value: 'OrderByPriceDESC',
+  },
+  ORDER_BY_PRICE_ASC: {
+    name: 'admin/editor.productSummaryList.orderType.priceAsc',
+    value: 'OrderByPriceASC',
+  },
+  ORDER_BY_NAME_ASC: {
+    name: 'admin/editor.productSummaryList.orderType.nameAsc',
+    value: 'OrderByNameASC',
+  },
+  ORDER_BY_NAME_DESC: {
+    name: 'admin/editor.productSummaryList.orderType.nameDesc',
+    value: 'OrderByNameDESC',
+  },
+  ORDER_BY_RELEASE_DATE_DESC: {
+    name: 'admin/editor.productSummaryList.orderType.releaseDate',
+    value: 'OrderByReleaseDateDESC',
+  },
+  ORDER_BY_BEST_DISCOUNT_DESC: {
+    name: 'admin/editor.productSummaryList.orderType.discount',
+    value: 'OrderByBestDiscountDESC',
+  },
+}
+
+const productQueryOptions = {
+  options: ({
+    category = '',
+    collection,
+    hideUnavailableItems = false,
+    orderBy = ORDER_BY_OPTIONS.ORDER_BY_TOP_SALE_DESC.value,
+    specificationFilters = [],
+    maxItems = 10,
+  }) => ({
+    ssr: true,
+    name: 'productList',
+    variables: {
+      category,
+      ...(collection != null
+        ? {
+            collection,
+          }
+        : {}),
+      specificationFilters: specificationFilters.map(parseFilters),
+      orderBy,
+      from: 0,
+      to: maxItems - 1,
+      hideUnavailableItems,
+    },
+  }),
+}
+
+function getOrdinationNames() {
+  const names = []
+  for (const key in ORDER_BY_OPTIONS) {
+    names.push(ORDER_BY_OPTIONS[key].name)
+  }
+  return names
```

Comment:
Maybe you could make a more abstract function like
```suggestion
function getOrdinationProp(propName) {
  return Object.keys(ORDER_BY_OPTIONS).map(key => ORDER_BY_OPTIONS[key][propName])
}
```

or you could generate both functions (`getOrdinationNames` and `getOrdinationValues`)

```javascript
function getOrdinationPropFn(propName) {
  return () => Object.keys(ORDER_BY_OPTIONS).map(key => ORDER_BY_OPTIONS[key][propName])
}

const getOrdinationNames = getOrdinationPropFn('name')
const getOrdinationValues = getOrdinationPropFn('value')
```

Reply:
Nice! Much cleaner